Auto Generate Document translator SDK

vikaspalaskar opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Commit:
Auto Generate Docuement translator SDK

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.AI.Translator.Document](https://apiview.dev/Assemblies/Review/13b038b18c3640a28f606c5136682d5e?revisionId=d217e8e514d645b9a8f2f138aa649247)

Commit:
Add hero scenario test.

Commit:
Update the http message for multipart.

Commit:
Supress methods

Commit:
Add CodeGenSuppress

Commit:
Update model name DocumentTranslateContent => multipartRequestContent

Commit:
move all methods to custom client

Commit:
Create document content

chunyu3 commented on sdk/translation/Azure.AI.Translator.Document/src/Custom/MultipartRequestContent.cs:
```diff
@@ -0,0 +1,86 @@
+        /// </summary>
+        internal IList<BinaryData> Glossary { get; }
+
+        internal MultipartFormDataContent MultipartFormDataContent { get; }
```
Hello please it is better not to add `MultipartFormDataContent` in the model. You can construct a multipartFormDataContent in the serialization method and convert it to RequestContent.

chunyu3 commented on sdk/translation/Azure.AI.Translator.Document/src/Custom/MultipartRequestContent.cs:
```diff
@@ -0,0 +1,86 @@
+
+namespace Azure.AI.Translator.Document
+{
+    public partial class DocumentContent
```
The filename is not consistent with the class name

vikaspalaskar committed:
Merge branch 'Azure:main' into main

Commit:
update models

Commit:
Merge branch 'main' of https://github.com/vikaspalaskar/azure-sdk-for-net

Commit:
refactor

- remove empty classes only for suppressing generated codes
- remove unused IUtf8JsonSerializable implementation
- add a new model for multiepart form file data
- update `DocumentContent` to use the model above
- lazy composing `MultipartFormDataContent`
- rename `tsp-location.yaml` to `tsp-location1.yaml` to work around a generator bug
- update test cases

archerzz commented on sdk/translation/Azure.AI.Translator.Document/tsp-location1.yaml:
```diff
@@ -0,0 +1,4 @@
+commit: ecf717be299977f1899bfea11a73f6c723698689
```
I have to rename this file so as to cheat the CI to not build this SDK. The reason is that I introduced a `MultipartFormFileData` which is used as type of the properties of the customized model. CSharp generator has an issue when handling custom models indirectly used.

m-nash commented on sdk/translation/Azure.AI.Translator.Document/src/Custom/MultipartFormFileData.cs:
```diff
@@ -0,0 +1,43 @@
+        {
+            Argument.AssertNotNull(name, nameof(name));
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNull(contentType, nameof(contentType));
```
this can be optional but should be defaulted to `application/octect-stream`

m-nash commented on sdk/translation/Azure.AI.Translator.Document/src/Custom/MultipartFormFileData.cs:
```diff
@@ -0,0 +1,43 @@
+        /// <param name="contentType">Content type of the file.</param>
+        public MultipartFormFileData(string name, BinaryData content, string contentType)
+        {
+            Argument.AssertNotNull(name, nameof(name));
```
Validate with the service team if this is optional or not.  I heard it can be optional?

vikaspalaskar replied:
I believe it is okay be to compulsory.

Commit:
update according to comments

- remove unused methods
- update doc comments
- add optional headers property to MultipartFormFileData
- optimize codes according to comments
- update api signature

Commit:
fix invalid link

archerzz commented on sdk/translation/Azure.AI.Translator.Document/src/Custom/MultipartFormFileData.cs:
```diff
@@ -0,0 +1,43 @@
+using System;
+using Azure.Core;
+
+namespace Azure.AI.Translator.Document
```
Discussed in the meeting. Pending on the sync-up meeting with Azure Core team.

lirenhe commented on sdk/translation/Azure.AI.Translator.Document/tests/DocumentClientTest.cs:
```diff
@@ -0,0 +1,62 @@
+            string filePath = Path.Combine("TestData", "test-input.txt");
+            using Stream fileStream = File.OpenRead(filePath);
+
+            var translateContent = new DocumentContent(new MultipartFormFileData(Path.GetFileName(filePath), BinaryData.FromStream(fileStream), "text/html"));
```
So to confirm, the content type defined here won't impact the format of the file which is always binary.

Commit:
Update the methods with document and glossaries parameter

Commit:
Update the SDK as per the SDK board recommendation

Commit:
Update the tests

Commit:
Update the tests

Commit:
Add doc tests

Commit:
Fix the build issues.

Commit:
add the classes

Commit:
remove method ToRequestContent

chunyu3 commented:
/azp run

chunyu3 commented:
/azp run net - translation - ci

Commit:
Fix the test issues

Commit:
Add asset file

Commit:
Update the tests as per the recommendation

Commit:
Update the test resource

Commit:
Add the latest asset file.

Commit:
disable tests due to this issue => https://github.com/Azure/azure-sdk-for-net/issues/41674

vikaspalaskar committed:
Merge branch 'Azure:main' into main

Commit:
Add model reference
<PackageReference Include="System.ClientModel" />

Commit:
regenerate the sdk as per the latest changes

Commit:
Use latest commit

Commit:
Remove Azure.AI.Translator.Document.netstandard2.0.cs

Commit:
Update the custom code so that it aligns with auto generated code.

Commit:
Update the tests and assets

Commit:
Add the tag => net/translation/Azure.AI.Translator.Document_9413b7ead9

Commit:
Make protocol method as public

Commit:
Move the internal method

chunyu3 commented on sdk/translation/Azure.AI.Translator.Document/src/Custom/MultipartFormFileData.cs:
```diff
@@ -0,0 +1,43 @@
+    /// <summary>
+    /// Represent a multipart form file data.
+    /// </summary>
+    public class MultipartFormFileData
```
DocumentContent is the body type of the operation. MultipartFormFileData is the property type of DocumentContent. So the serialize logic still need to be in DocumentContent class. I think
